components/script/dom/htmlmediaelement.rs

```diff
@@ -1234,6 +1246,12 @@ impl HTMLMediaElement {
             PlayerEvent::FrameUpdated => {
                 self.upcast::<Node>().dirty(NodeDamage::OtherNodeDamage);
             },
+            PlayerEvent::RateChanged(r) => {
+                if self.playbackRate.get() != r {
+                    self.playbackRate.set(r);
+                    self.queue_ratechange_event();
+                }
```

Comment:
Let's do this slightly different to avoid duplicated `ratechange` events:

```rust
if self.playbackRate.get() == r {
    self.queue_ratechange_event();
} else {
    self.playbackRate.set(r);
}
```

Reply:
Which tests are timing out? Could you update the PR with your latest changes, please?

If it is timing out it may be because we are not getting the `Player::RateChanged` in some cases or because there is something else wrong.

Reply:
Here are the tests that time out:
![image](https://user-images.githubusercontent.com/30772943/50099440-d067ff00-0226-11e9-82f0-43eaba11f46a.png)

Reply:
It seems that the problem is in servo-media. Setting the playback rate [here](https://github.com/servo/media/blob/master/backends/gstreamer/src/player.rs#L392) completely breaks a/v playback. It even [failed on Travis](https://travis-ci.org/servo/media/jobs/467472875). Sorry, I should have catched that on the review.

I still don't know why that fails. I will investigate more tomorrow.

Reply:
I found the reason why this was not working. We were creating a deadlock while trying to send the RateChanged event. The good news is that we don't need this event at all. I mistakenly thought that the rate change signal was triggered when the actual rate change was effective and that the player could change its playback rate for other reasons other than the client asking for that change, but apparently none of my assumptions were correct. The signal is triggered only when the rate **property** is changed and the playback rate will only change if the client instructs the player to change it and so we don't need to notify the client about that change.

I am removing the signal handler and the Player::RateChanged event in https://github.com/servo/media/pull/175.

Reply:
@georgeroman https://github.com/servo/media/pull/175 has already landed, so we can update this PR to remove the Player::RateChanged logic. Thanks for your pacience.

Reply:
Great! I updated the code accordingly and now the tests pass.